core/src/Kokkos_Core.hpp

```diff
@@ -149,7 +149,7 @@ namespace Kokkos {
 
 namespace Impl {
 
-static inline void check_init_final([[maybe_unused]] char const* func_name) {
+inline void check_init_final([[maybe_unused]] char const* func_name) {
```

Comment:
any idea why this was static?

Reply:
> any idea why this was static?

No. It was introduced in https://github.com/kokkos/kokkos/pull/7676.